src/python/pants/engine/goal.py

```diff
@@ -46,7 +46,12 @@ def activated(cls, union_membership: UnionMembership) -> bool:
 
     @classmethod
     def create_scope_info(cls, **scope_info_kwargs) -> ScopeInfo:
-        return super().create_scope_info(is_goal=True, **scope_info_kwargs)
+        return super().create_scope_info(
+            # `filter` should now be treated as a subsystem for `help`, even though it
+            # still works as a goal for backwards compatibility.
+            is_goal=False if cls.name == "filter" else True,
```

Comment:
We should make this declarative, otherwise this is waaaaay too hidden and bound to grow

Reply:
I was considering doing that, but felt like too much indirection. I'm having a hard time envisioning other cases where specifically a GoalSubsystem now becomes a normal Subsystem...For example, `update-build-files` will be removed or deprecated, but it's not going to become a global subsystem usable by any goal. `filter` is a unique duckling